sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -977,6 +983,10 @@ def upload_blob(self, repository: str, data: IO[bytes], **kwargs) -> Tuple[str,
                     **kwargs
                 )
             )
+            if digest != complete_upload_response_headers["Docker-Content-Digest"]:
+                raise ManifestDigestValidationException(
+                    "The server-computed digest does not match the client-computed digest."
+                )
```

Comment:
Following the digest validation in other languages. @lmolkova since the validation is after the `complete` call, do you know if we need to delete the blob when digest is invalid?